docs/docs/porting-a-jekyll-website-to-gatsby.md

```diff
@@ -0,0 +1,276 @@
+---
+title: Porting a Jekyll Site to Gatsby
+---
+
+# Jekyll
+
+[Jekyll](https://jekyllrb.com/) is a static site generator for your websites. Think of it like a file-based CMS. It takes your content, renders Markdown, Liquid templates, and spits out a complete, static website ready to be served by [Apache](https://www.apache.org/), [Nginx](https://www.nginx.com/) or another web server. In Jekyll if you write a markdown file in the root folder under `_site` thay will considered `pages`, meanwhile if you write `.md` files in `_posts` folder they will considered `posts` that's how jekyll works.
+
+## What changes need to be made?
+
+In order to transition your codebase over to using Gatsby, a few things need to be taken care of to account for the differences between how the projects are set up. First of all Gatsby want to know where to read `markdown` files by using the plugin [`gatsby-source-filesystem`](https://www.gatsbyjs.org/packages/gatsby-source-filesystem/) `( source plugin for sourcing data into your Gatsby application)` but how does it understand markdown format for this we need to use the plugin [`gatsby-transformer-remark`](https://www.gatsbyjs.org/packages/gatsby-transformer-remark/) to transform markdown files into something like gatsby can understand. This plugin is the responsible to read `markdown` files a extract information like the `frontmatter`, the excerpt and convert the `text` into `HTML` ready to be rendered.
```

Comment:
```suggestion
In order to transition your codebase over to using Gatsby, a few things need to be taken care of to account for the differences between the project setups. 

First, Gatsby needs to know the expected location of your markdown files. This is configured using the [`gatsby-source-filesystem`](https://www.gatsbyjs.org/packages/gatsby-source-filesystem/) plugin, a source plugin for markdown data. 

Additionally, Gatsby uses the [`gatsby-transformer-remark`](https://www.gatsbyjs.org/packages/gatsby-transformer-remark/) plugin to transform markdown files into something that Gatsby can understand. This plugin is responsible for reading markdown files and extracting information like [frontmatter](/docs/adding-markdown-pages/#frontmatter-for-metadata-in-markdown-files).
```

Reply:
Wondering if this section is necessary. It is helpful for Gatsby internals context but I'd expect more of an overview at this point in the doc.

Perhaps the sentences about above `_site` and `_posts` should move down here? And then a more general note about how Gatsby uses plugins to locate and transform markdown?